File: twoSum.py
'''
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

You can return the answer in any order.
'''

def twoSum(array,value):
    band = False
    for i in array:
        diff = value - i
        if diff in array:
            band = True
            return band
    return band

File: cpp/code/twoSum.cpp
/*
Given two lists of integers, is there a set of numbers — one from each list — whose sum is equal to a specified value
*/

#include <iostream>
#include <vector>
using namespace std;

bool twoSum(vector<int> list, int value);

bool twoSum(vector<int> list, int value)
{
	for (int i = 0; i < list.size(); i++)
	{
		int diff = value - i;
		for (int j = i + 1; j < list.size(); j++)
		{
			if (list[j] == diff)
			{
				return true;
			}
		}
	}
	return false;
}